src/main/java/gregtech/api/pipenet/block/BlockPipe.java

```diff
@@ -354,6 +351,17 @@ public boolean onPipeActivated(World world, IBlockState state, BlockPos pos, Ent
             }
         }
 
+        if (itemStack.getItem().getToolClasses(itemStack).contains(ToolClasses.CROWBAR)) {
+            if (!world.isRemote) {
+                pipeTile.getCoverableImplementation().removeCover(coverSide);
```

Comment:
This functions returns `boolean` if it was successful. So you should work with it.

Reply:
The only case that this function could return false is if there is not a cover at the passed in side, which is something that is already checked for earlier in the method, therefore this would always return true.

Reply:
Yes, but if someone extracts this part to own function later use it here and reuse it somewhere else and does not check code 20 lines above it will generate bug. Same goes if some changes that code above without checking this part.

In all cases I would go with defensive programing. And add here that one if.